core/src/test/groovy/com/novoda/gradle/release/JavaDifferentGradleVersions.groovy

```diff
@@ -0,0 +1,51 @@
+package com.novoda.gradle.release
+
+import com.novoda.gradle.release.rule.TestProjectRule
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+import static org.assertj.core.api.Assertions.assertThat
+
+@RunWith(Parameterized.class)
+class JavaDifferentGradleVersions {
```

Comment:
you could combine this with the Android test class so that the tests are all in the same place - makes it muc easier to understand / maintain

Reply:
I'm not sure is a good idea @blundell, as we need to create different synthetic projects and test separate outcomes too.

Reply:
I was scrolling between the two test classes and couldn't see much difference. Is there big setup overhead? Seems like they have some connescence with change was my thinking

Reply:
to achieve what you suggest we need to treat the projects as another dimension, that would male the test harder to understand and maintain IMO. I'd be happy with some duplication here.

Reply:
I'm 50% with you. The code is more or less copied code which could be merged.

On the other side, the bintray-release plugin provides two different features. One is the **creating** of `jars` / `aars`. The other is the publishing to bintray... (more or less).
Only because of that reason it make sense to me to have two different test classes which tests two different possible scenarios.

It also brings much more complexity into our tests which can be avoided if we have two different test classes...